weave/ops_arrow/list_ops.py

```diff
@@ -675,15 +676,25 @@ def concat(arr):
     elif len(arr) == 1:
         return arrow_tags.pushdown_list_tags(arr[0])
 
-    res = arr[0]
-    res = typing.cast(ArrowWeaveList, res)
-    res = arrow_tags.pushdown_list_tags(res)
-
-    for i in range(1, len(arr)):
-        tagged = arrow_tags.pushdown_list_tags(arr[i])
-        res = res.concat(tagged)
-    return res
+    tagged = list(builtin_map(lambda x: arrow_tags.pushdown_list_tags(x), arr))
+    return merge_concat(tagged)
```

Comment:
not familiar with arrowweavelist, can we pre-allocate and do this in 1 pass?

Reply:
I believe we should be able to. But I'm about as familiar (or even less) with AWL as you are, so I didn't attempt that for now.

Reply:
Tabling the 1-pass investigation for now due to technical complications. See Slack discussion [here](https://weightsandbiases.slack.com/archives/C03BSTEBD7F/p1687281960989229?thread_ts=1686765095.853029&cid=C03BSTEBD7F).